docs/parameters.md

```diff
@@ -0,0 +1,141 @@
+# Passing parameters to brokers
+
+Table of Contents
+- [Overview](#overview)
+- [Design](#design)
+  - [Basic example](#basic-example)
+  - [Passing sensitive data](#passing-sensitive-data)
+  - [Merging multiple sources and conflict resolution](#merging-multiple-sources-and-conflict-resolution)
+- [Example with multiple sources](#example-with-multiple-sources)
+
+## Overview
+`parameters` and `parametersFrom` properties of `Instance` and `Broker` resources 
+provide support for passing parameters to the broker relevant to the corresponding
+[provisioning](https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#provisioning) or
+[binding](https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#binding) request. 
+The resulting structure represents an arbitrary JSON object, which is assumed to 
+be valid for a particular broker. 
+The Service Catalog does not enforce any extra limitations on the format and content 
+of this structure.
+
+## Design
+
+When you create an `Instance` or a `Binding`, you can set parameters to be passed 
+to the corresponding broker.
+To set parameters, include the `parameters` or `parametersFrom` field in the spec.
```

Comment:
Perhaps:
- - - -
To set parameters, you may use the `parameters` and `parametersFrom` fields in the `Spec` of the Instance or Binding:
- `parameters` : can be used to specify a set of properties to be sent to the broker. The data specified will be passed "as-is" to the broker without any modifications - aside from converting it to json for transmission to the broker in the case of the Spec being specified as yaml. Any valid yaml or json constructs are supported. One only `parameters` fields may be specified per Spec.
- `parametersFrom` : can be used to specify which secret, and key in that secret, contains a `string` that represents the json to include in the set of parameters to be sent to the broker. Multiple `parametersFrom` fields may be specified per Spec.

You may use either, or both, of these fields as needed.

If multiple sources in `parameters` and `parametersFrom` blocks are specified,
the final payload is a result of merging all of them at the top level.
If there are any duplicate properties defined at the top level, the specification
is considered to be invalid, the further processing of the `Instance`/`Binding` 
resource stops and its `status` is marked with error condition.

The format of the Spec will be:
```
spec:
  ...
  parameters:
    name: value
  parametersFrom:
    - secretKeyRef:
        name: secretName
        key: myKey
```
or
```
"spec": {
  ...
  "parameters": {
    "name": "value"
  },
  "parametersFrom": {
    "secretKeyRef": {
      "name": "secretName",
      "key": "myKey"
    }
  }
}
```
and the secret would need to have a key named `myKey`:
```
apiVersion: v1
kind: Secret
metadata:
  name: secretName
type: Opaque
stringData:
  myKey: >
    {
      "property": "value"
    }
```
- - - -
it might read nicer if all of the descriptive stuff was in one spec instead of split across two sections.